go.mod

```diff
@@ -147,4 +149,4 @@ exclude (
 
 // Go modules keeps resetting the required v0.35.0 to a random unversioned commit.
 // So this is required for unknown reasons.
-replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.35.0
+replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.41.0
```

Comment:
nit: curious if this is still necessary

Reply:
What makes you curious? We need the version bump because of the API changes to build with Prometheus 2.41. Unless you're suggesting we could update other Prometheus client libraries?

Reply:
This `replace` directive was put in as a hack because go modules was doing something funky. In theory, we should be able to point to `v0.41.0` above where the original Prometheus [dependency](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/fa652aea7a407d3528e06d80c7df20171ac8683f/go.mod#L19) is configured.

Reply:
Double checked this!

Looks like without the replace directive, after you run `go mod tidy`, you get `github.com/prometheus/prometheus v1.8.2-0.20211119115433-692a54649ed7` which confirms you still need the directive (sadly).